readthedocs/settings/base.py

```diff
@@ -402,6 +393,81 @@ def MIDDLEWARE(self):
     ]
     PYTHON_MEDIA = False
 
+    # Content Security Policy
+    # https://django-csp.readthedocs.io/
+    CSP_FRAME_ANCESTORS = ("'none'",)
+    CSP_OBJECT_SRC = ("'none'",)
+    CSP_REPORT_URI = None
+    CSP_REPORT_ONLY = False
+    CSP_EXCLUDE_URL_PREFIXES = ("/admin/",)
+
+    # Default to disallow everything, and then allow specific sources on each directive.
+    CSP_DEFAULT_SRC = ["'none'"]
+    CSP_SCRIPT_SRC = [
+        "'self'",
+        STATIC_URL,
+        # Some of our JS deps are using eval.
+        "'unsafe-eval'",
+        # Allow fontawesome to load.
+        "https://kit.fontawesome.com",
+        # Stripe (used for Gold subscriptions)
+        "https://js.stripe.com/",
+    ]
+    CSP_CONNECT_SRC = [
+        "'self'",
+        # Allow sentry to report errors.
+        "https://*.ingest.us.sentry.io",
+        # Allow fontawesome to load.
+        "https://ka-p.fontawesome.com",
+        "https://kit.fontawesome.com",
+        # Plausible analytics
+        "https://plausible.io/api/event",
+    ]
+    CSP_IMG_SRC = [
+        "'self'",
+        # Some of our styles include images as data URLs.
+        "data:",
+        # We load avatars from GitHub, GitLab, and Bitbucket,
+        # and other services. They don't use a single specific domain,
+        # so we just allow any https domain here.
+        "https:",
```

Comment:
Just a note, that whenever we use https, it should be http for development. So probably those settings should be dynamic, or maybe could be extended in the settings used for dev to add a http rule.

Reply:
It seems this is to load images from GH and similar. I'm not hitting an issue locally leaving `https:` here.